File: week1/24.swap_nodes_in_pairs/24.swap_nodes_in_pairs.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swapPairs(head *ListNode) *ListNode {
	dummy := &ListNode{}
	dummy.Next = head
	tmp := dummy

	for tmp.Next != nil && tmp.Next.Next != nil {
		node1, node2 := tmp.Next, tmp.Next.Next

		tmp.Next = node2
		node1.Next = node2.Next
		node2.Next = node1
		tmp = node1
	}
	return dummy.Next
}

File: Algorithms/24.swap_nodes_in_pairs-两两交换链表中的节点/24.swap_nodes_in_pairs.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
 var swapPairs = function(head) {
     let dummy = new ListNode();
     dummy.next = head;
     let tmp = dummy;

     while (tmp.next != null && tmp.next.next != null) {
         let node1 = tmp.next;
         let node2 = tmp.next.next;

         tmp.next = node2;
         node1.next = node2.next;
         node2.next = node1;
         tmp = node1;
     }
     return dummy.next;
};